sihl_users/lib/service.ml

```diff
@@ -209,20 +209,25 @@ module User = struct
     in
     if not @@ Model.Token.is_valid_email_configuration token then
       Sihl_core.Fail.raise_bad_request "invalid confirmation token provided"
-      (* TODO do this with transaction *)
     else
-      let* () =
-        Repository.Token.update (Model.Token.inactivate token)
-        |> Sihl_core.Db.query_db_exn request
-      in
-      let* user =
-        Repository.User.get ~id:token.user |> Sihl_core.Db.query_db request
+      let* result =
+        Sihl_core.Db.query_db_with_trx request (fun connection ->
+            let* () =
+              Repository.Token.update (Model.Token.inactivate token)
+              |> Sihl_core.Db.query_db_exn request
+            in
+            let* user =
+              Repository.User.get ~id:token.user
+              |> Sihl_core.Db.query_db request
+            in
+            let user =
+              user |> Sihl_core.Fail.with_bad_request "invalid token provided"
+            in
+            Repository.User.update (Model.User.confirm user) connection)
```

Comment:
@jerben Out of curiosity: When are connections closed?

Reply:
@aronerben Very good point! We need to stress test the whole thing to find leaking connections. All connections are retrieved from a connection pool so they should be managed by it. We don't drain it atm anywhere, the end of a Sihl application life cycle hasn't been implemented really yet.

Reply:
When I close the connection on the MariaDB driver it segfaults even, on Postgres it works.

Reply:
So it's a Caqti problem?

Reply:
@aronerben I don't know yet. I think it is wrong usage to release a connection while we retrieved it using `Pool.use (fun connection -> ...)`. The underlying bindings are directly wrapping the C drivers, maybe there is something wrong.